src/main/java/io/javalin/Javalin.java

```diff
@@ -166,7 +177,9 @@ public Javalin stop() {
         eventManager.fireEvent(JavalinEvent.SERVER_STOPPING);
         log.info("Stopping Javalin ...");
         try {
-            jettyServer.stop();
+            if (jettyServer != null) {
```

Comment:
I think a nullpointer here would be better than nothing.

Reply:
stop() is generally expected to do nothing if the server is not started; throwing npe is unexpected.

Reply:
Then `if (jettyServer == null) throw new AppropriateException()`. If we can extend Javalin and override start/stop the problem solves itself.

Reply:
Passing `jettyServer` and `jettySessionHandler` as constructor parameters seem to work pretty nicely, so I was able to remove the now-useless null check.